examples/statemachine/src/language-server/main-browser.ts

```diff
@@ -15,7 +15,21 @@ const messageWriter = new BrowserMessageWriter(self);
 const connection = createConnection(messageReader, messageWriter);
 
 // Inject the shared services and language-specific services
-const { shared } = createStatemachineServices({ connection, ...EmptyFileSystem });
+const { shared, statemachine } = createStatemachineServices({ connection, ...EmptyFileSystem });
 
 // Start the language server with the shared services
 startLanguageServer(shared);
+
+// Send a notification with the serialized AST after every document change
+type DocumentChange = { uri: string, content: string };
+const documentChangeNotification = new NotificationType<DocumentChange>('browser/DocumentChange');
+const jsonSerializer = statemachine.serializer.JsonSerializer;
+shared.workspace.DocumentBuilder.onBuildPhase(DocumentState.Validated, documents => {
+    for (const document of documents) {
+        const json = jsonSerializer.serialize(document.parseResult.value);
+        connection.sendNotification(documentChangeNotification, {
```

Comment:
What about errors in the AST? Do you also want the user aware of validation, syntax and lexer errors?

Reply:
Those are already sent here:
https://github.com/langium/langium/blob/e24dee8a4085e520c8ac319e836caa6c596fb6ac/packages/langium/src/lsp/language-server.ts#L206-L221

Reply:
Oki doki!